source/hp/dof_handler.cc

```diff
@@ -1506,6 +1506,30 @@ namespace hp
 
 
 
+  template <int dim, int spacedim>
+  void
+  DoFHandler<dim, spacedim>::execute_coarsening_and_refinement()
+  {
+    for (const auto &cell : active_cell_iterators())
+      if (cell->is_locally_owned())
```

Comment:
Do we need to check for artificial cells? So far all tests pass without one...

Reply:
I don't think you need to set flags on `ghost` cells.

Reply:
Hmm. I guess I should extend the domain in those tests.

Right now there is always exactly one processor who owns the whole domain. The others just store one artificial cell on the top level.

Reply:
Seems to work even with this extension. I guess we're good.